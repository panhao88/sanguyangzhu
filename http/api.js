const fly = require('./index')

export default {
  // 首页轮播图
  tBanner(current, size) {
    return fly.get(`/tBanner?current=${current}&size=${size}`)
  },
  // 首页分类信息
  tClassification(current, size) {
    return fly.get(`/tClassification?current=${current}&size=${size}`)
  },
  // 首页资讯
  tInformation(current, size) {
    return fly.get(`/tInformation?current=${current}&size=${size}`)
  },
  //养殖基地
  tPigFarmManage(current, size) {
    return fly.get(`/tPigFarmManage?current=${current}&size=${size}`)
  },
  // 授权登录请求 存在邀请码
  loginInCode(code, parentInCode) {
    return fly.get(`/login/getAccessToken/` + code + `?parentInCode=${parentInCode}`)
  },
  // 授权登录请求 不存在邀请码
  login(code) {
    return fly.get('/login/getAccessToken/' + code)
  },
  // 存在邀请码
  updateIncode(parentInCode, data) {
    return fly.put(`/tUser?parentInCode=${parentInCode}`, data)
  },
  // 不存在邀请码
  update(data) {
    return fly.put(`/tUser`, data)
  },
  // 获取用户信息
  userinfo() {
    return fly.get(`/tUser`)
  },
  // 邀请记录
  popularize(id) {
    return fly.get(`/popularize/getInviteDetail/{id}?id=${id}`)
  },
  // 推广首页
  getpopularize(id) {
    return fly.get(`/popularize/${id}`)
  },
  // 销售业绩
  getAchievement(id) {
    return fly.get(`/popularize/getAchievement/${id}`)
  },
  //获取商品列表
  tGoods(data) {
    if (data.id !== undefined) {
      return fly.get(`/tGoods?current=${data.current}&size=${data.size}&goodsTypeId=${data.id}`)
    } else {
      return fly.get(`/tGoods?current=${data.current}&size=${data.size}`)
    }
  },
  // 产品详情
  tGoodsDetail(id) {
    return fly.get(`/tGoodsDetail/${id}`)
  },
  // 个人中心 常见问题
  tQuestion(current, size) {
    return fly.get(`/tQuestion?current=${current}&size=${size}`)
  },
  // 个人中心  关于我们
  tAbout(id) {
    return fly.get(`/tAbout/${id}`)
  },
  //累计收益记录
  tTotalIncomeDetait(id, current, size) {
    return fly.get(`/tTotalIncome?userId=${id}&current=${current}&size=${size}`)
  },
  //养猪收益记录
  tPigIncome(id, current, size) {
    return fly.get(`/tPigIncome?userId=${id}&current=${current}&size=${size}`)
  },
  // 猪粮资产记录
  tIntegralAssets(id, current, size) {
    return fly.get(`/tIntegralAssets?userId=${id}&current=${current}&size=${size}`)
  },
  //销售奖金记录
  tSaleMoney(id, current, size) {
    return fly.get(`/tSaleMoney?userId=${id}&current=${current}&size=${size}`)
  },
  //提现记录
  tWithdrawalRecord(id, current, size) {
    return fly.get(`/tWithdrawalRecord?userId=${id}&current=${current}&size=${size}`)
  },
  // 我的钱包
  TotalIncome(id) {
    return fly.post(`tTotalIncome/myWallet?userId=${id}`)
  },
  // 围观伙伴
  tWatch(current, size, ppId) {
    return fly.get(`/tWatch?current=${current}&size=${size}&ppId=${ppId}`)
  },
  // 个人中心  消息中心
  tMessagePush(current, size, userId) {
    return fly.get(`/tMessagePush?current=${current}&size=${size}&userId=${userId}`)
  },
  // 验证码请求
  sendMessage(phone) {
    return fly.get(`/tIdentity/send?phone=${phone}`)
  },
  //  获取所有省的信息
  tProvinces(current, size) {
    return fly.get(`/tProvinces?current=${current}&size=${size}`)
  },
  //获取城市
  getCity(provinceid) {
    return fly.get(`/tCities/getCityOrArea?provinceid=${provinceid}`)
  },
  //获取城市
  getArea(cityid) {
    return fly.get(`/tCities/getCityOrArea?cityid=${cityid}`)
  },
  //获取用户身份位/tUser/selectUserIdentity
  getIdentity(userId) {
    return fly.get(`/tUser/selectUserIdentity?userId=${userId}`)
  },
  // 实名认证/tIdentity
  tIdentity(phone, code, name, area, idCard, idCardFace, idCardNational, userId) {
    return fly.post(`/tIdentity?phone=${phone}&code=${code}&name=${name}&area=${area}&idCard=${idCard}&idCardFace=${idCardFace}&idCardNational=${idCardNational}&userId=${userId}`)
  },
  // 获取销售奖金
  achievement(id) {
    return fly.get(`/tOrder/achievement?id=${id}`)
  },
  // 我的订单
  tOrder() {
    return fly.get(`/tOrder`)
  },
  //查询所有订单 findAll
  getOrder(current, size, userId) {
    return fly.get(`/tOrder/findAll?current=${current}&size=${size}&userId=${userId}`)
  },
  // 未支付
  tOrder(current, size, userId, state) {
    return fly.get(`/tOrder?current=${current}&size=${size}&userId=${userId}&state=${state}`)
  },
  // 认养订单
  adoptTOrder(current, size, userId, orderType) {
    return fly.get(`/tOrder?current=${current}&size=${size}&userId=${userId}&orderType=${orderType}`)
  },
  // 转让订单
  transfertOrder(current, size, userId, orderType) {
    return fly.get(`/tOrder?current=${current}&size=${size}&userId=${userId}&orderType=${orderType}`)
  },
  //订单详情
  orderDetail(goodsId, orderType, userId) {
    return fly.post(`/tOrder/orderDetail?goodsId=${goodsId}&userId=${userId}&orderType=${orderType}`)
  },
  // 首页系统消息推送
  tMessagePush(current, size, userId) {
    return fly.get(`/tMessagePush?current=${current}&size=${size}&userId=${userId}`)
  },
  // 产品价格
  tGoodsInfo(id) {
    return fly.get(`/tGoodsInfo/show?id=${id}`)
  },
  // 我的金猪
  pigtOrder(current, size, userId, ) {
    return fly.get(`/tOrder?current=${current}&size=${size}&userId=${userId}`)
  },
  // 产品类型
  tGoodsType(current, size) {
    return fly.get(`/tGoodsType?current=${current}&size=${size}`)
  },
  // 产品条件查询
  tGoodsFindAll(goodsTypeId) {
    return fly.get(`/tGoods/findAll?goodsTypeId=${goodsTypeId}`)
  },
  // 母猪信息
  tParentPig(id){
    return fly.get(`/tParentPig/${id}`)
  },
  // 按胎次查询
  tChildPig(ppId){
    return fly.get(`/tChildPig/count/${ppId}`)
  },
   // 根据胎次获得猪崽信息
   getTChilpigByParity(parity,ppId){
    return fly.get(`/tChildPig/getTChilpigByParity?parity=${parity}&ppId=${ppId}`)
   },
  //  消杀防疫
  tDisinfect(current,size){
    return fly.get(`/tDisinfect?current=${current}&size=${size}`)
  },
  // 市场全部的猪
  tSalePig(current,size){
    return fly.get(`/tSalePig?current=${current}&size=${size}`)
  },
    // 市场猪分类  金猪
    getSalePig(current,size,type){
      return fly.get(`/tSalePig?current=${current}&size=${size}&type=${type}`)
    },
}