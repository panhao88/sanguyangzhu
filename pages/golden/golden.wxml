<!--pages/golden/golden.wxml-->
<view>
  <!-- 顶部导航 -->
  <!-- <view class='nav bg-white' style='height:{{navH}}px'>
    <view class='nav-title'>
      <image src="../../images/image/fanhui.png" class="return" bindtap="return"></image>
      我的金猪
    </view>
  </view> -->
  <my-nav title="我的金猪">
    <view>
      <image src='../../images/image/changtu.png' class='back' style='height:{{navH}}px'></image>
      <!-- 第一个 -->
      <!-- <view class="kuan" wx:for="{{pig}}" wx:key="index">
        <view class="dier">耳标号:{{item.earNumber}}</view>
        <view class="box">
          <view class="dif" bindtap="information">
            <view class="beijing">
              <image src="{{item.picture}}" class="zhu"></image>
            </view>
            <view class="shu">
              <view class="san">{{item.name}}</view>
              <view class="yipai">
                <view>周期:</view>
                <view class="color">{{item.period}}个月</view>
              </view>
            </view>
          </view>
          <view class="onclick">
            <view class="mai" bindtap="onClickShow">我要卖猪</view>
            <view class="watch" bindtap="watch">邀请围观</view>
          </view>
        </view>
      </view> -->
      <!-- 点击我要卖猪弹框 -->
      <van-overlay show="{{ show }}">
        <view class="wrapper">
          <view class="block">
            <view class="tishi">提示</view>
            <view class="ipt">
              <view class="pri">价格</view>
              <input type="text" placeholder="您的理想价格" class="de" />
            </view>
            <view class="flexs">
              <view class="cancel" bindtap="onClickHide">取消</view>
              <view class="release">发布订单</view>
            </view>
          </view>
        </view>
      </van-overlay>
      <!-- 第二个 -->
      <view class="boxs" wx:for="{{pig}}" wx:key="index">
        <view class="dier">耳标号:{{item.earNumber}}</view>
        <view class="map">
          <view class="dif" data-state="{{item.state}}" data-id="{{item.ppId}}" bindtap="number">
            <view class="beijing">
              <image src="{{item.picture}}" class="zhu"></image>
            </view>
            <view class="shu">
              <view class="san">{{item.name}}</view>
              <view class="yipai">
                <view>周期:</view>
                <view class="color">{{item.period}}个月</view>
              </view>
            </view>
          </view>
          <view wx:if="{{item.state === 4}}">
            <view class="mai" bindtap="onClickShow">我要卖猪</view>
            <view class="invitation" bindtap="watch">邀请围观</view>
          </view>
          <view wx:else>
            <view class="mai" data-state="{{item.state}}" data-id="{{item.ppId}}" bindtap="number">
              {{item.state===0?'未支付':(item.state===1?'已支付':(item.state===2?'猪苗采购中':(item.state===3?'等待选号':(item.state===4?'已完成':(item.state===5?'已退款':'')))))}}
            </view>
          </view>
        </view>
      </view>
      <!-- <view class="boxs" wx:for="{{pig}}" wx:key="index">
        <view class="map">
          <view class="dif">
            <view class="beijing">
              <image src="http://sangu-miniapp.oss-cn-hangzhou.aliyuncs.com/test/images/1612423422029.jpg" class="zhu"></image>
            </view>
            <view class="shu">
              <view class="san">{{item.name}}</view>
              <view class="yipai">
                <view>周期:</view>
                <view class="color">{{item.period}}个月</view>
              </view>
            </view>
          </view>
          <view class="mai">猪苗采购中</view>
        </view>
      </view> -->
    </view>
  </my-nav>
</view>