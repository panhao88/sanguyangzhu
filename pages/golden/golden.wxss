/* pages/golden/golden.wxss */
page {
  height: 100%;
  background: #f1f1f1;
}

.nav {
  width: 100%;
  height: 45rpx;
  overflow: hidden;
  position: relative;
  top: 0;
  left: 0;
  z-index: 10;
}

.nav-title {
  width: 100%;
  height: 45rpx;
  line-height: 45rpx;
  text-align: center;
  position: absolute;
  bottom: 30rpx;
  left: 0;
  z-index: 10;
  font-size: 36rpx;
  font-family: PingFang SC;
  font-weight: bold;
  color: #FFFFFF;
}

.return {
  width: 30rpx;
  height: 45rpx;
  position: absolute;
  left: 32rpx;
}

.back {
  width: 750rpx;
  height: 128rpx;
  position: absolute;
  top: 0;
  left: 0;
  z-index: -1;
}

.wrapper {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
}

.block {
  width: 580rpx;
  height: 347rpx;
  background: #FFFFFF;
  border-radius: 20rpx;
  display: flex;
  align-items: center;
  flex-direction: column;
}

.tishi {
  font-size: 36rpx;
  font-family: PingFang SC;
  font-weight: bold;
  color: #222222;
  margin-top: 46rpx;
}

.ipt {
  width: 492rpx;
  height: 80rpx;
  background: #EFEFEF;
  border-radius: 6rpx;
  margin-top: 45rpx;
  display: flex;
  align-items: center;
}
.pri {
font-size: 28rpx;
font-family: PingFang SC;
font-weight: bold;
color: #333333;
margin-left: 25rpx;
}
.de {
margin-left: 35rpx;
font-size: 28rpx;
font-family: PingFang SC;
font-weight: 400;
color: #898989;
}
.flexs {
  display: flex;
  margin-top: 32rpx;
}
.cancel {
  width: 230rpx;
  height: 78rpx;
  border: 2px solid #EFEFEF;
  border-radius: 10rpx;
  display: flex;
  justify-content: center;
  align-items: center;
font-size: 28rpx;
font-family: PingFang SC;
font-weight: bold;
color: #7A7A7A;
margin-right: 25rpx;
}
.release {
  width: 230rpx;
  height: 78rpx;
  background: #6CA543;
  border-radius: 10rpx;
  display: flex;
  justify-content: center;
  align-items: center;
font-size: 28rpx;
font-family: PingFang SC;
font-weight: bold;
color: #FFFFFF;
}
.kuan {
  width: 686rpx;
  height: 224rpx;
  background: #FFFFFF;
  box-shadow: 0px 0px 18rpx 0px rgba(0, 0, 0, 0.1);
  border-radius: 20rpx;
  margin-left: 32rpx;
  margin-top: 26rpx;
}

.dier {
  font-size: 26rpx;
  font-family: PingFang SC;
  font-weight: 500;
  color: #494949;
  margin-top: 23rpx;
  margin-left: 28rpx;
  padding-top: 26rpx;
}

.beijing {
  width: 126rpx;
  height: 126rpx;
  background: #E6E5EB;
  border-radius: 26rpx;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 24rpx;
  margin-left: 26rpx;
}

.zhu {
  width: 126rpx;
  height: 126rpx;
  border-radius: 26rpx;
}

.box {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.dif {
  display: flex;
  align-items: center;
}

.shu {
  display: flex;
  flex-direction: column;
  margin-left: 30rpx;
}

.san {
  font-size: 30rpx;
  font-family: PingFang SC;
  font-weight: 500;
  color: #494949;
  margin-top: 24rpx;
}

.yipai {
  display: flex;
  font-size: 28rpx;
  font-family: PingFang SC;
  font-weight: 500;
  color: #898989;
  line-height: 50rpx;
  margin-top: 10rpx;
}

.color {
  color: #6CA543;
}

.mai {
  width: 170rpx;
  height: 60rpx;
  background: #6CA543;
  border-radius: 20rpx;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 28rpx;
  font-family: PingFang SC;
  font-weight: 500;
  color: #FFFFFF;
  line-height: 50rpx;
  margin-right: 26rpx;
}
.invitation {
  width: 170rpx;
  height: 60rpx;
  background: #6CA543;
  border-radius: 20rpx;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 28rpx;
  font-family: PingFang SC;
  font-weight: 500;
  color: #FFFFFF;
  line-height: 50rpx;
  margin-right: 26rpx;
  margin-top: 26rpx;
}
.onclick {
  display: flex;
  flex-direction: column;
}
.watch {
  width: 170rpx;
  height: 60rpx;
  background: #6CA543;
  border-radius: 20rpx;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 28rpx;
  font-family: PingFang SC;
  font-weight: 500;
  color: #FFFFFF;
  line-height: 50rpx;
  margin-right: 26rpx;
  margin-top: 26rpx;
}
.boxs {
  width: 686rpx;
  height: 224rpx;
  background: #FFFFFF;
  box-shadow: 0px 0px 18rpx 0px rgba(0, 0, 0, 0.1);
  border-radius: 20rpx;
  margin-left: 32rpx;
  margin-top: 26rpx;
  /* display: flex;
  justify-content: space-between;
  align-items: center; */
}
.map {
  width: 686rpx;
  display: flex;
  justify-content: space-between;
  align-items: center;
}