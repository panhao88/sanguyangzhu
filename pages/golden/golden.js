// pages/golden/golden.js
const app = getApp()
import api from "../../http/api"
Page({

  /**
   * 页面的初始数据
   */
  data: {
    current:1,
    size:10,
    show: false,
    pig:[], //我的金猪
  },
  // 点击返回键返回上一级
  return () {
    wx.navigateBack({
      delta: 1
    })
  },
  // 点击我要卖猪弹框
  onClickShow() {
    this.setData({
      show: true
    });
  },
  onClickHide() {
    this.setData({
      show: false
    });
  },
  // 点击邀请围观进入页面
  watch() {
    wx.navigateTo({
      url: '/pages/watch/watch',
    })
  },
  // 我的金猪数据请求
  pigtOrder(){
    api.pigtOrder(this.data.current,this.data.size,wx.getStorageSync('uid')).then(res=>{
      console.log(res);
      this.setData({
        pig:res.data.records
      })
    })
  },
  // 点击  state： 0未支付 1已支付  2猪苗采购中 3等待选号 4已完成 5已退款 
  number(e){
    console.log(e);
    let id = e.currentTarget.dataset.id
    let state = e.currentTarget.dataset.state
    if(state === 0){

    }else if(state === 1){

    }else if(state === 2){

    }else if(state === 3){
      wx.navigateTo({
        url: '/pages/label/label',
      })
    }else if(state === 4){
      wx.navigateTo({
        url: `/pages/information/information?id=${id}`,
      })
    }
  },
  /**
   * 生命周期函数--监听页面加载
   */
  onLoad: function (options) {
    this.pigtOrder()
    // 顶部自定义标题
    // console.log(app.globalData.navHeight)
    this.setData({
      navH: app.globalData.navHeight
    })
  },

  /**
   * 生命周期函数--监听页面初次渲染完成
   */
  onReady: function () {

  },

  /**
   * 生命周期函数--监听页面显示
   */
  onShow: function () {

  },

  /**
   * 生命周期函数--监听页面隐藏
   */
  onHide: function () {

  },

  /**
   * 生命周期函数--监听页面卸载
   */
  onUnload: function () {

  },

  /**
   * 页面相关事件处理函数--监听用户下拉动作
   */
  onPullDownRefresh: function () {

  },

  /**
   * 页面上拉触底事件的处理函数
   */
  onReachBottom: function () {

  },

  /**
   * 用户点击右上角分享
   */
  onShareAppMessage: function () {

  }
})