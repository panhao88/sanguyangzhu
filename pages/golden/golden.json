{
  "usingComponents": {
    "van-overlay": "/lib/vant/overlay",
    "my-nav":"/components/nav/nav"
  }
}